pkg/resource/certificate/sdk.go

```diff
@@ -0,0 +1,642 @@
+			}
+			if r.ko.Spec.APIPassthrough.Subject.SerialNumber != nil {
+				f0f1.SetSerialNumber(*r.ko.Spec.APIPassthrough.Subject.SerialNumber)
+			}
+			if r.ko.Spec.APIPassthrough.Subject.State != nil {
+				f0f1.SetState(*r.ko.Spec.APIPassthrough.Subject.State)
+			}
+			if r.ko.Spec.APIPassthrough.Subject.Surname != nil {
+				f0f1.SetSurname(*r.ko.Spec.APIPassthrough.Subject.Surname)
+			}
+			if r.ko.Spec.APIPassthrough.Subject.Title != nil {
+				f0f1.SetTitle(*r.ko.Spec.APIPassthrough.Subject.Title)
+			}
+			f0.SetSubject(f0f1)
+		}
+		res.SetApiPassthrough(f0)
+	}
+	if r.ko.Spec.CertificateAuthorityARN != nil {
+		res.SetCertificateAuthorityArn(*r.ko.Spec.CertificateAuthorityARN)
+	}
+	if r.ko.Spec.CSR != nil {
+		res.SetCsr(r.ko.Spec.CSR)
+	}
+	if r.ko.Spec.IdempotencyToken != nil {
+		res.SetIdempotencyToken(*r.ko.Spec.IdempotencyToken)
+	}
+	if r.ko.Spec.SigningAlgorithm != nil {
+		res.SetSigningAlgorithm(*r.ko.Spec.SigningAlgorithm)
+	}
+	if r.ko.Spec.TemplateARN != nil {
+		res.SetTemplateArn(*r.ko.Spec.TemplateARN)
+	}
+	if r.ko.Spec.Validity != nil {
+		f6 := &svcsdk.Validity{}
+		if r.ko.Spec.Validity.Type != nil {
+			f6.SetType(*r.ko.Spec.Validity.Type)
+		}
+		if r.ko.Spec.Validity.Value != nil {
+			f6.SetValue(*r.ko.Spec.Validity.Value)
+		}
+		res.SetValidity(f6)
+	}
+	if r.ko.Spec.ValidityNotBefore != nil {
+		f7 := &svcsdk.Validity{}
+		if r.ko.Spec.ValidityNotBefore.Type != nil {
+			f7.SetType(*r.ko.Spec.ValidityNotBefore.Type)
+		}
+		if r.ko.Spec.ValidityNotBefore.Value != nil {
+			f7.SetValue(*r.ko.Spec.ValidityNotBefore.Value)
+		}
+		res.SetValidityNotBefore(f7)
+	}
+
+	return res, nil
+}
+
+// sdkUpdate patches the supplied resource in the backend AWS service API and
+// returns a new resource with updated fields.
+func (rm *resourceManager) sdkUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	return nil, ackerr.NewTerminalError(ackerr.NotImplemented)
+}
+
+// sdkDelete deletes the supplied resource in the backend AWS service API
+func (rm *resourceManager) sdkDelete(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkDelete")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newDeleteRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+	var resp *svcsdk.RevokeCertificateOutput
+	_ = resp
+	resp, err = rm.sdkapi.RevokeCertificateWithContext(ctx, input)
+	rm.metrics.RecordAPICall("DELETE", "RevokeCertificate", err)
+	return nil, err
+}
+
+// newDeleteRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Delete API call for the resource
+func (rm *resourceManager) newDeleteRequestPayload(
+	r *resource,
+) (*svcsdk.RevokeCertificateInput, error) {
+	res := &svcsdk.RevokeCertificateInput{}
+
+	if r.ko.Spec.CertificateAuthorityARN != nil {
+		res.SetCertificateAuthorityArn(*r.ko.Spec.CertificateAuthorityARN)
+	}
+
```

Comment:
Not sure but aren't we missing the serial of the certificate here? looks like it's a required value

Reply:
https://docs.aws.amazon.com/privateca/latest/APIReference/API_RevokeCertificate.html#privateca-RevokeCertificate-request-CertificateSerial

Reply:
Yes CertificateSerial and RevocationReason are required for RevokeCertificate, I just added them as Spec fields